cockatrice/src/game/player/player.cpp

```diff
@@ -1341,30 +1342,43 @@ void Player::actMoveTopCardsUntil()
 {
     stopMoveTopCardsUntil();
 
-    DlgMoveTopCardsUntil dlg(game, previousMovingCardsUntilExpr, previousMovingCardsUntilNumberOfHits);
+    DlgMoveTopCardsUntil dlg(game, movingCardsUntilExpr, movingCardsUntilNumberOfHits, movingCardsUntilAutoPlay);
     if (!dlg.exec()) {
         return;
     }
 
-    previousMovingCardsUntilExpr = dlg.getExpr();
-    previousMovingCardsUntilNumberOfHits = dlg.getNumberOfHits();
+    movingCardsUntilExpr = dlg.getExpr();
+    movingCardsUntilNumberOfHits = dlg.getNumberOfHits();
+    movingCardsUntilAutoPlay = dlg.isAutoPlay();
 
     if (zones.value("deck")->getCards().empty()) {
         stopMoveTopCardsUntil();
     } else {
-        movingCardsUntilFilter = FilterString(previousMovingCardsUntilExpr);
-        movingCardsUntilCounter = previousMovingCardsUntilNumberOfHits;
+        movingCardsUntilFilter = FilterString(movingCardsUntilExpr);
+        movingCardsUntilCounter = movingCardsUntilNumberOfHits;
         movingCardsUntil = true;
         actMoveTopCardToPlay();
     }
 }
 
-void Player::moveOneCardUntil(const CardInfoPtr card)
+void Player::moveOneCardUntil(CardItem *card)
 {
     moveTopCardTimer->stop();
-    if (zones.value("deck")->getCards().empty() || card.isNull()) {
```

Comment:
you seem to have removed the null check, does this work?

Reply:
fixed